File: Data Structures & Algorithms/duplicate-integer/submission-0.go
func hasDuplicate(nums []int) bool {
    mapper := make(map[int]bool)

    for _, v := range nums {
        _ , ok := mapper[v]
        if ok {
            mapper[v] = true
            return true
        } else {
            mapper[v] = false
        }
    }
    return false
}

File: Data Structures & Algorithms/missing-number/submission-0.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        ans ={}

        for x in nums:
            if x in ans:
                ans[x] +=1
            else:
                ans[x] = 1
        
        for i in range(1000000000000000000000):
            if i not in ans:
                return i
        